keystrokes/src/components/Character/CharacterDetails/ManualKeybindModal.js

```diff
@@ -0,0 +1,251 @@
+	allKeybindings: PropTypes.object.isRequired,
+	keyBindingKey: PropTypes.string.isRequired,
+};
+
+function ManualKeybindModal({
+	isOpen,
+	setIsOpen,
+	characterClass,
+	characterSpec,
+	setAllKeybindings,
+	allKeybindings,
+	keyBindingKey,
+}) {
+	const classes = styleGuide();
+	const [keybinding, setKeybinding] = useState({
+		Spell: null,
+		Target: null,
+		Mod: null,
+		Key: null,
+	});
+	const [allKeybinds, setAllKeybinds] = useState([]);
+	const [loading, setLoading] = useState(true);
+	const [Spells, setSpells] = useState();
+	const spec = characterDetails.class[characterClass][
+		characterSpec
+	].toUpperCase();
+
+	useEffect(() => {
+		async function getSpells() {
+			const path = `/Spells/${characterClass}`;
+			const snapShot = await ref.child(path).once('value');
+			if (snapShot.exists()) {
+				setSpells(snapShot.val());
+				setLoading(false);
+			}
+		}
+		getSpells();
+	}, []);
+
+	function handleKeyPress(e) {
+		if (!e.metaKey) {
+			e.preventDefault();
+		}
+		const newKey = e.code.toLowerCase().replace(/digit|key/i, '');
+		if (validatePress(newKey)) {
+			setKeybinding({
+				...keybinding,
+				Key: verifyKey(newKey),
+			});
+		} else if (newKey === 'backspace' || newKey === 'delete') {
+			setKeybinding({
+				...keybinding,
+				Key: '',
+			});
+		}
+	}
+
+	return (
+		<Modal open={isOpen} onClose={() => {}} className={classes.modal}>
+			<div className={classes.manualModalBackground}>
+				{loading ? (
+					<div>Loading</div>
+				) : (
+					<React.Fragment>
+						<Grid container justify="space-between">
+							<Grid item>
+								<Button
+									color="secondary"
+									variant="contained"
+									onClick={() => {
+										setAllKeybinds([]);
+										setIsOpen(false);
+									}}
+									size="large"
+								>
+									Cancel
+								</Button>
+							</Grid>
+							<Grid item>
+								<Button
+									color="primary"
+									variant="contained"
+									size="large"
+									onClick={() => {
+										setAllKeybindings(
+											update(allKeybindings, {
+												[keyBindingKey]: {
+													$push: allKeybinds,
+												},
+											})
+										);
+										setAllKeybinds([]);
+										setIsOpen(false);
+									}}
+								>
+									Finish
+								</Button>
+							</Grid>
+						</Grid>
+						<Grid
+							container
+							className={classes.paddingTop}
+							justify="center"
+							alignItems="center"
+						>
+							<Grid item className={classes.keybindingOptions}>
+								<TextField
+									className={classes.button}
+									select
+									variant="outlined"
+									value={keybinding.Spell || ''}
+									label={'Spell'}
+									onChange={event => {
+										setKeybinding({
+											...keybinding,
+											Spell: event.target.value,
+										});
+									}}
+								>
+									{Object.entries(Spells)
+										.filter((spell, index) => {
+											return spell[1].spec.includes(spec);
```

Comment:
doesn't look like index is used in the filter 
could also use short hand => spell[1].spec...